flexget/components/sites/sites/piratebay.py

```diff
@@ -75,11 +91,12 @@ def set_urls(self, url):
         if self.url != url:
             self.url = url
             parsed_url = urlparse(url)
+            # most api mirrors share the same domain with frontend, except thepiratebay.org which uses apibay.org
             self.url_match = re.compile(
-                r'^%s://(?:torrents\.)?(%s)/.*$'
+                r'^%s://(?:[a-z0-9]+\.)?(?:thepiratebay\.org(?:\:\d+)?|%s)/description\.php\?id=(\d+)$'
```

Comment:
Does this regex need to be this specific to work? It seems potentially fragile to me.

Reply:
Because the API domain is now not always the same with site domain (for instance the official thepiratebay.org uses apibay.org). Valid URLs for rewriting are thepiratebay.org and the URL set in the config. I need to tweak this as config URL might contain path.